janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -1277,8 +1278,12 @@ else if (query.getResultType() == ElementCategory.VERTEX) {
                         }
                     });
                 }
+                //we construct an iterator which lazily streams results from 1st index, and filters by looking up in the intersection of results from all other indices (if any)
+                //NOTE we need to pass NO_LIMIT to processIntersectingRetrievals, otherwise the intersection will be cut down to limit and potentially miss correct results
+                //so this only makes sense if 1st index is the one returning most results (and supporting streaming) AND the rest of indexes are
```

Comment:
This is a great explanation for the fix and helpful for reviewing the change, but won't make sense once the change us applied. Please remove this line.